fv3config/_asset_list.py

```diff
@@ -276,7 +276,8 @@ def check_asset_has_required_keys(asset):
 def config_to_asset_list(config):
     """Convert a configuration dictionary to an asset list. The asset list
     will contain all files for the run directory except the namelist."""
-    asset_list = get_initial_conditions_asset_list(config)
```

Comment:
for certain conditions, this expression resolves to `asset_list = config["initial_conditions"]` in which case this entry in the config dict gets modified by later modifications of the `asset_list`